ecosystem/typescript/sdk/src/aptos_client.test.ts

```diff
@@ -534,3 +534,25 @@ test(
   },
   longTestTimeout,
 );
+
+test(
+  "view function",
+  async () => {
+    const client = new AptosClient(NODE_URL);
+    const faucetClient = getFaucetClient();
+
+    const alice = new AptosAccount();
+    await faucetClient.fundAccount(alice.address(), 100_000_000);
+
+    const payload: Gen.ViewRequest = {
+      function: "0x1::coin::balance",
+      type_arguments: ["0x1::aptos_coin::AptosCoin"],
+      arguments: [alice.address().hex()],
+    };
+
+    const balance = await client.view(payload);
+
+    expect(balance[0]).toBe("100000000");
```

Comment:
What does the return value look like for complicated returned types such as structs?

Reply:
Do we allow structs as outputs of view functions?  They have to be droppable right?

Also, if we do, I believe it would be formatted the same as any other structs from the API, similar to listing resources.

Reply:
View functions can return structs and they don't have to be droppable currently actually. We're planning to add more conditions in the future.

Reply:
I think it'd be great to add a test for this case and other cases for completeness. Try https://github.com/aptos-labs/aptos-core/blob/main/aptos-move/framework/aptos-framework/sources/configs/staking_config.move#L92

Reply:
That's not required I believe